test/puppetlabs/ssl_utils/core_test.clj

```diff
@@ -500,6 +501,60 @@
             (is (revoked? updated-crl cert))
             (is (revoked? updated-crl cert2))))))))
 
+(deftest verify-crls-test
+ (testing "verify CRLs against CA cert chain"
+   (let [cert-chain (-> "crl_validation/3-cert-chain.pem" open-ssl-file pem->certs)
+         crl-chain (-> "crl_validation/3-crl-chain.pem" open-ssl-file pem->crls)
+         other-crl-chain (-> "crl_validation/other-3-crl-chain.pem" open-ssl-file pem->crls)
+         crl-chain-missing-crls (-> "crl_validation/2-crl-chain.pem" open-ssl-file pem->crls)
+         crl-chain-extra-crls (-> "crl_validation/4-crl-chain.pem" open-ssl-file pem->crls)
+         cert-with-expired-crl (-> "crl_validation/cert-with-expired-crl.pem" open-ssl-file pem->certs)
+         expired-crl (-> "crl_validation/expired-crl.pem" open-ssl-file pem->crls)
+         cert-with-not-valid-crl (-> "crl_validation/cert-with-not-valid-crl.pem" open-ssl-file pem->certs)
+         not-yet-valid-crl (-> "crl_validation/not-yet-valid-crl.pem" open-ssl-file pem->crls)
+         cert-chain-with-crl-bad-sig (-> "crl_validation/cert-with-crl-bad-sig.pem" open-ssl-file pem->certs)
+         crl-chain-bad-signature (-> "crl_validation/crl-with-bad-signature.pem" open-ssl-file pem->crls)
+         cert-chain-with-revoked-cert (-> "crl_validation/cert-chain-with-revoked-cert.pem" open-ssl-file pem->certs)
+         crl-chain-cert-revoked (-> "crl_validation/crl-chain-with-cert-revoked.pem" open-ssl-file pem->crls)
+         ;;crl-chain-delta (open-ssl-file "crl_validation/3-crl-chain-delta.pem")
+         ]
+     (testing "valid certs and CRL chain"
+       (is (nil? (validate-cert-chain cert-chain crl-chain))))
+     (testing "extra CRLs are okay"
+       (is (nil? (validate-cert-chain cert-chain crl-chain-extra-crls))))
+     (testing "missing CRLs are rejected"
+       ;; currently UNDETERMINED REVOCATION STATUS
+       (is (thrown-with-msg? CertPathValidatorException
+                             #"Could not determine revocation status"
+                             (validate-cert-chain cert-chain crl-chain-missing-crls))))
+     (testing "totally unrelated CRLs are rejected"
```

Comment:
This is testing when none of the CRLs match?

Reply:
yeah, it's just 3 CRLs from a totally different cert chain, with different keys and everything.

Reply:
which sort of falls under the category of 'missing CRLs' I guess.. but it seemed conceptually different.